### PR TITLE
roachtest: ignore flaky activerecord tests

### LINKED ISSUE #134811: roachtest: activerecord failed
**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17690126?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17690126?buildTab=artifacts#/activerecord) on release-24.2 @ [5764366875adca70c3a3ff3d4d13aaf599bc53a7](https://github.com/cockroachdb/cockroach/commits/5764366875adca70c3a3ff3d4d13aaf599bc53a7):

```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v24.2.5-dev-5764366875adca70c3a3ff3d4d13aaf599bc53a7
Tests run against activerecord 7.2.1
8491 Total Tests Run
8465 tests passed
26 tests failed
60 tests skipped
1 test ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=true</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17690126/activerecord/1731307421412/1731315700687)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133931 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.5-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44263

### LINKED ISSUE #135801: roachtest: activerecord failed
roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17851096?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17851096?buildTab=artifacts#/activerecord) on master @ [8eeb7f2ae3b2cede564b46ca47e2353fd147c061](https://github.com/cockroachdb/cockroach/commits/8eeb7f2ae3b2cede564b46ca47e2353fd147c061):

```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.1.0-alpha.00000000-dev-8eeb7f2ae3b2cede564b46ca47e2353fd147c061
Tests run against activerecord 7.2.1
8491 Total Tests Run
8467 tests passed
24 tests failed
60 tests skipped
1 test ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17851096/activerecord/1732084923853/1732093362699)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135548 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #134811 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.2]
- #133931 roachtest: activerecord failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.2.5-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44720

### PR DESCRIPTION
**Pull request by annrpom**
Fixes: #134811
Fixes: https://github.com/cockroachdb/cockroach/issues/135801

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/activerecord_blocklist.go [hunk 1]
```diff
@@ -18,29 +18,32 @@ package tests
...
+	`CompatibilityTest6_1#test_options_are_not_validated`:                                                                                                      "unknown",
+	`CompatibilityTest7_0#test_datetime_sets_precision_6_on_change_column`:                                                                                     "unknown",
+	`CompatibilityTest7_0#test_options_are_not_validated`:                                                                                                      "unknown",
+	`PostGISTest#test_point_to_json`:                                                                                                                           "unknown",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update`: "unknown",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason`:                             "unknown",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not`:            "unknown",
```
Comment (rafiss) [inline pkg/cmd/roachtest/tests/activerecord_blocklist.go:46]: nit: these new tests are flaky, so should be in the ignoreList.

(right now they are in the blockList, which means the test will expect them to fail every time.)

**Review by rafiss — CHANGES_REQUESTED** [review]
i just assigned https://github.com/cockroachdb/cockroach/issues/135801 to you as well, would you be able to add that test too if it's not too much trouble?

so the full list of tests to ignore:
```
TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason
TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not
TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update

CockroachDB::AdapterForeignKeyTest#test_foreign_key_violations_are_translated_to_specific_exception_with_validate_false
CockroachDB::AdapterForeignKeyTest#test_foreign_key_violations_on_insert_are_translated_to_specific_exception
CockroachDB::AdapterForeignKeyTest#test_foreign_key_violations_on_delete_are_translated_to_specific_exception

BasicsTest#test_default_values_are_deeply_dupped
````

**Comment by annrpom** [comment]
> let's hold off on ignoring those until we see that they are more unstable

ok i'll keep my eye out for those!

File: pkg/cmd/roachtest/tests/activerecord_blocklist.go (hunk 1 above)
Comment (annrpom) [reply pkg/cmd/roachtest/tests/activerecord_blocklist.go]: ah apologies

File: pkg/cmd/roachtest/tests/activerecord_blocklist.go [hunk 2]
```diff
@@ -44,5 +44,8 @@ var activeRecordBlocklist = blocklist{
 }
 
 var activeRecordIgnoreList = blocklist{
-	`ActiveRecord::ConnectionAdapters::PostgreSQLAdapterTest#test_translate_no_connection_exception_to_not_established`: "pg_terminate_backend not implemented",
+	`ActiveRecord::ConnectionAdapters::PostgreSQLAdapterTest#test_translate_no_connection_exception_to_not_established`:                                        "pg_terminate_backend not implemented",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update`: "flaky",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason`:                             "flaky",
+	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not`:            "flaky",
```
Comment (rafiss) [inline pkg/cmd/roachtest/tests/activerecord_blocklist.go:50]: can we add `BasicsTest#test_default_values_are_deeply_dupped`?

**Review by rafiss — CHANGES_REQUESTED** [review]

File: pkg/cmd/roachtest/tests/activerecord_blocklist.go (hunk 2 above)
Comment (annrpom) [reply pkg/cmd/roachtest/tests/activerecord_blocklist.go:51]: my reading comprehension skills were afoot

**Review by rafiss — APPROVED** [review]
lgtm!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135907#-:-OCL6Tao2KI_olTP34q5:b-7nbn20)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nameisbhaskar and @vidit-bhat)

**Comment by annrpom** [comment]
TFTR! ('-')7

ty for adding those backport labels

bors r+

**Pull request merged**